tests/integration_tests/functional/test_config_space.py

```diff
@@ -0,0 +1,351 @@
+    """Test changing the MAC address of the network device."""
+    global PAYLOAD_DATA_SIZE
+
+    test_microvm = test_microvm_with_ssh
+    test_microvm.spawn()
+    test_microvm.basic_config(boot_args="ipv6.disable=1")
+
+    # Data exchange interface ('eth0' in guest).
+    iface_id = '1'
+    _tap1, host_ip1, guest_ip1 = test_microvm.ssh_network_config(
+        network_config,
+        iface_id
+    )
+
+    # Control interface ('eth1' in guest).
+    iface_id = '2'
+    _tap2, _, guest_ip2 = test_microvm.ssh_network_config(
+        network_config,
+        iface_id
+    )
+
+    # Configure metrics, to get later the `tx_spoofed_mac_count`.
+    metrics_fifo_path = os.path.join(test_microvm.path, 'metrics_fifo')
+    metrics_fifo = log_tools.Fifo(metrics_fifo_path)
+    response = test_microvm.metrics.put(
+        metrics_path=test_microvm.create_jailed_resource(metrics_fifo.path)
+    )
+    assert test_microvm.api_session.is_status_no_content(response.status_code)
+
+    test_microvm.start()
+
+    # Create the control ssh connection.
+    test_microvm.ssh_config['hostname'] = guest_ip2
+    ssh_connection_ctl = net_tools.SSHConnection(test_microvm.ssh_config)
+
+    # Start a server(host) - client(guest) communication with the following
+    # parameters.
+    host_port = 4444
+    iterations = 1
+    _exchange_data(
+        test_microvm.jailer,
+        ssh_connection_ctl,
+        host_ip1,
+        host_port,
+        iterations
+    )
+
+    fc_metrics = _flush_metrics(test_microvm, metrics_fifo)
+    assert fc_metrics['net']['tx_spoofed_mac_count'] == 0
+
+    # Change the MAC address of the network data interface.
+    # This change will be propagated only inside the net device kernel struct
+    # and will be used for ethernet frames formation when data is exchanged
+    # on the network interface.
+    mac = "06:05:04:03:02:01"
+    mac_hex = "0x060504030201"
+    guest_if1_name = _get_net_if_name(ssh_connection_ctl, guest_ip1)
+    assert guest_if1_name is not None
+    _change_guest_if_mac(ssh_connection_ctl, mac, guest_if1_name)
+
+    _exchange_data(
+        test_microvm.jailer,
+        ssh_connection_ctl,
+        host_ip1,
+        host_port,
+        iterations
+    )
+
+    # `tx_spoofed_mac_count` metric was incremented due to the MAC address
+    # change.
+    fc_metrics = _flush_metrics(test_microvm, metrics_fifo)
+    assert fc_metrics['net']['tx_spoofed_mac_count'] > 0
+
+    if platform.machine() == "x86_64":
+        net_addr_base = _get_net_mem_addr_base_on_x86_64(ssh_connection_ctl,
+                                                         guest_if1_name)
+    elif platform.machine() == "aarch64":
+        net_addr_base = _get_net_mem_addr_base_on_aarch64(ssh_connection_ctl,
+                                                          guest_if1_name)
+    else:
+        assert False
+
+    assert net_addr_base is not None
+
+    # Write into '/dev/mem' the same mac address, byte by byte.
```

Comment:
meta-nit: is there a virtio-friendly way to trigger this without `/dev/mem`? Mmap-ing `/dev/mem` and writing some bytes at some address is not exactly the first thing that comes to mind when thinking about changing a MAC address.